src/coreclr/gc/gc.cpp

```diff
@@ -19863,7 +19863,14 @@ start_no_gc_region_status gc_heap::prepare_for_no_gc_region (uint64_t total_size
     }
 
     int soh_align_const = get_alignment_constant (TRUE);
+#ifdef USE_REGIONS
+    // With regions, we 'could' extend the SOH as large as we wanted by chaining the
+    // regions together. However, it probably does not make sense to have a huge SOH
+    // the next ephemeral GC will suffer badly otherwise.
+    size_t max_soh_allocated = 256 * 1024 * 1024;
```

Comment:
I would not have this limit at all. it's fine if the next ephemeral GC is very long as this is a very special case and the user should be well aware that they might incur a long GC right after. 

the reason why I had a limit before was simply due to the fact that our SOH segment was always of the same size (since our tuning is depending on the segment size it'd require more work to have an SOH segment that could be way larger than other segments). with regions we don't have this concern so I would not have enforce a limit for gen0 allocations.